File: src/java/com/github/resttest/http/matchers/RegexMatcher.java
package com.github.resttest.http.matchers;

import org.hamcrest.Description;
import org.hamcrest.TypeSafeMatcher;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class RegexMatcher extends TypeSafeMatcher<String> {

    private final String regex;

    public RegexMatcher(final String regex) {
        this.regex = regex;
    }

    @Override
    public void describeTo(final Description description) {
        description.appendText("Regular expression = \"" + regex + "\"");
    }

    @Override
    public boolean matchesSafely(final String string) {
        Matcher m = Pattern.compile(regex).matcher(string);
        if (m.find()) return true;
        return false;
    }


    // matcher classMethod you can call on this matcher class
    public static RegexMatcher regex(final Object regex) {
        return new RegexMatcher((String) regex);
    }
}

File: src/main/kotlin/com/github/aivancioglo/resttest/matchers/RegexMatcher.kt
package com.github.aivancioglo.resttest.matchers

import org.hamcrest.Description
import org.hamcrest.Factory
import org.hamcrest.TypeSafeMatcher
import java.util.regex.Pattern.compile

@Deprecated("Use RestTestMatcher class")
class RegexMatcher(private val regex: String) : TypeSafeMatcher<String>() {
    override fun describeTo(description: Description) {
        description.appendText("Regular expression = \"$regex\"")
    }

    override fun matchesSafely(text: String) = compile(regex).matcher(text).find()

    companion object {
        @Factory
        @JvmStatic
        @Deprecated("Use regex in RestTestMatcher class")
        fun regex(regex: Any) = RegexMatcher(regex as String)
    }
}
